feat: Improve ResourceManager UX (#9338)
ipfs/kubo: docs/config.md

### `Swarm.ResourceMgr`

The [libp2p Network Resource Manager](https://github.com/libp2p/go-libp2p-resource-manager#readme) allows setting limits per a scope,
and tracking recource usage over time.

#### `Swarm.ResourceMgr.Enabled`

Enables the libp2p Network Resource Manager and auguments the default limits
using user-defined ones in `Swarm.ResourceMgr.Limits` (if present).

Various `*rcmgr_*` metrics can be accessed as the prometheus endpoint at `{Addresses.API}/debug/metrics/prometheus` (default: `http://127.0.0.1:5001/debug/metrics/prometheus`)

Default: `true`

Type: `flag`

> #### `Swarm.ResourceMgr.MaxMemory`

Reviewer: A couple of thoughts:
1. Should we use this type: https://github.com/ipfs/kubo/blob/master/docs/config.md#optionalbytes ?
2. If not, would it make sense to put the unit in the name to make it self documenting like we do in https://github.com/ipfs/kubo/blob/master/docs/config.md#internalbitswapmaxoutstandingbytesperpeer
3. If we are specifying a units, maybe choose MB instead of bytes (and then adjust code accordingly).

That said, item 1 seems best from a user perspective.

> Define the maximum amount of memory used by libp2p.

Reviewer 2: I would be even more specific, there are lots of things in libp2p that consume memory that are not tracked by RM (also applies to MaxFileDescriptors below)

```suggestion
The maximum amount of memory that the libp2p resource manager will allow.
```

I also like OptionalBytes

> Define the maximum amount of memory used by libp2p.

Author: moved to OptionalBytes (it is basically an OptionalString being parsed by humanize)

> Default: `[TOTAL_MEMORY]/8`

Reviewer: ```suggestion
Default: `[TOTAL_SYSTEM_MEMORY]/8`
```

I don't know the best term for this but am trying to come up with a clear term.

> Type: `integer64`

Reviewer: Per above, can this be https://github.com/ipfs/kubo/blob/master/docs/config.md#optionalbytes

#### `Swarm.ResourceMgr.MaxFileDescriptors`

> Define the maximum amount of file descriptors used by libp2p.

Reviewer: ```suggestion
Define the maximum number of file descriptors that libp2p can use.
```

> Default `[TOTAL_FILE_DESCRIPTORS]/2`

Reviewer: ```suggestion
Default `[TOTAL_SYSTEM_FILE_DESCRIPTORS]/2`
```

Should we add a comment on what happens in Windows cases?  I assume we'll follow https://github.com/libp2p/go-libp2p/search?q=getNumFDs and set it to infinite by default (or do we want to set to a different value).
